packages/mobile-web/src/tools/mobile-offline/offline-analysis/tool.ts

```diff
@@ -0,0 +1,230 @@
+  readonly toolId = 'sfmobile-web-offline-analysis';
+  readonly inputSchema = LwcCodeSchema;
+  readonly outputSchema = ExpertsCodeAnalysisIssuesSchema;
+
+  private readonly linter: Linter;
+  private readonly ruleReviewers: Record<string, CodeAnalysisBaseIssueType>;
+
+  constructor() {
+    this.linter = new Linter({ configType: 'flat' });
+    this.ruleReviewers = this.initializeRuleReviewers();
+  }
+
+  public register(server: McpServer, annotations: ToolAnnotations): void {
+    server.registerTool(
+      this.toolId,
+      {
+        description: this.description,
+        inputSchema: this.inputSchema.shape,
+        annotations: annotations,
+      },
+      async (code: LwcCodeType) => {
+        try {
+          const analysisResults = await this.analyzeCode(code);
+          return {
+            content: [
+              {
+                type: 'text',
+                text: this.createAnalysisMarkdown(code.name, analysisResults),
+              },
+            ],
+          };
+        } catch (error) {
+          console.error('Error during code analysis:', error);
+          throw new Error(
+            `Failed to analyze code: ${error instanceof Error ? error.message : 'Unknown error'}`
+          );
+        }
+      }
+    );
+  }
+
+  private initializeRuleReviewers(): Record<string, CodeAnalysisBaseIssueType> {
+    return ruleConfigs.reduce(
+      (acc, ruleConfig) => {
+        acc[ruleConfig.id] = ruleConfig.config;
+        return acc;
+      },
+      {} as Record<string, CodeAnalysisBaseIssueType>
+    );
+  }
+
+  private async analyzeCode(code: LwcCodeType): Promise<ExpertsCodeAnalysisIssuesType> {
+    const jsCode = code.js.map(js => js.content).join('\n');
+    const { messages } = this.linter.verifyAndFix(jsCode, LINTER_CONFIG, {
+      fix: true,
+    });
+
+    const offlineAnalysisIssues = this.analyzeIssues(jsCode, messages);
+
+    return {
+      analysisResults: [offlineAnalysisIssues],
+      orchestrationInstructions: this.getOrchestrationInstructions(),
+    };
+  }
+
+  private getOrchestrationInstructions(): string {
+    return "If you have not already executed the 'sfmobile-web-offline-guidance' tool in this conversation thread: (1) Immediately call the 'sfmobile-web-offline-guidance' tool, (2) Execute all review instructions from the guidance tool's output against the user's LWC component code in the specified format, (3) Combine the review results from step 2 with this tool's analysis results, then refactor the component based on all combined review recommendations.";
+  }
+
+  private createAnalysisMarkdown(
+    componentName: string,
+    analysisResults: ExpertsCodeAnalysisIssuesType
+  ): string {
+    let markdown = `Analysis results for the LWC component ${componentName} are presented below. Please review them and provide a detailed refactoring plan for the component. \n\n`;
```

Comment:
nitpick:   string concatenation is expensive.  use array to hold the individual, by the end do a array.join() to get the final result?